src/_pytest/setuponly.py

```diff
@@ -66,7 +67,11 @@ def _show_fixture_action(fixturedef, msg):
             tw.write(" (fixtures used: {})".format(", ".join(deps)))
 
     if hasattr(fixturedef, "cached_param"):
-        tw.write("[{}]".format(fixturedef.cached_param))
+        if isinstance(fixturedef.cached_param, bytes):
```

Comment:
Have you considered calling `saferepr` unconditionally? This will have the effect of adding quotes to strings, but that might be a good idea anyway.

Reply:
I did, but was too afraid to change the behavior for all types. Actually I agree - showing repr output here probably makes more sense.